api/apps/api/src/modules/scenarios/planning-units/scenario-planning-units-protection-status-calculator-service.ts

```diff
@@ -0,0 +1,46 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+
+import { DbConnections } from '@marxan-api/ormconfig.connections';
+import { ScenariosPlanningUnitGeoEntity } from '@marxan/scenarios-planning-unit';
+import { Scenario } from '../scenario.api.entity';
+
+@Injectable()
+export class ScenarioPlanningUnitsProtectedStatusCalculatorService {
+  constructor(
+    @InjectRepository(
+      ScenariosPlanningUnitGeoEntity,
+      DbConnections.geoprocessingDB,
+    )
+    private readonly puRepo: Repository<ScenariosPlanningUnitGeoEntity>,
+  ) {}
+
+  /**
+   * Calculate default lock status of planning units given their intersection
+   * with protected areas.
+   */
+  async calculatedProtectionStatusForPlanningUnitsIn(scenario: Scenario): Promise<void> {
+    const wdpaIucnCategoriesForScenario = scenario.wdpaIucnCategories?.map(i => `'${i}'`).join(', ');
+    console.log(wdpaIucnCategoriesForScenario);
+
+    const query = `
+    with pa as (select * from wdpa where iucn_cat in (${wdpaIucnCategoriesForScenario})),
+    pu as (
+    select spd.id, pug.the_geom
+    from scenarios_pu_data spd
+    inner join planning_units_geom pug on spd.pu_geom_id = pug.id
+    where scenario_id='${scenario.id}'),
+    pu_pa as (select pu.id, st_area(st_intersection(pu.the_geom, pa.the_geom)) as pa_pu_area, (CASE pu.the_geom && pa.the_geom WHEN true THEN 2 else 0 end) as lockin_status
```

Comment:
as we discussed yesterday we need to separate pu lock status from wdpa area calculation. in this query it was together

Reply:
sure - is this what you had in mind? ab93e4e0ebbd26b0a2617d0d4bd995b14157bafc

Reply:
this can be made easier as if we are separating wdpa area calculation from lockin status so i would suggest:
```query
with pu as (
    select spd.id,  (pug.protected_area/spd.area)*100 as perc_protection
    from scenarios_pu_data spd
    inner join planning_units_geom pug on spd.pu_geom_id = pug.id
    where scenario_id='${scenario.id}'),
    pu_pa as (select pu.id, (CASE pu.perc_protection > ${threshold} WHEN true THEN 2 else 0 end) as lockin_status
from pu) 
UPDATE scenarios_pu_data
    SET (lockin_status) =
        (SELECT lockin_status FROM pu_pa
         WHERE scenarios_pu_data.id = pu_pa.id) where scenario_id = '${scenario.id}'; 
```

Reply:
thanks! 965f0d80fabe9d6e83c0682342975fddfa934d63